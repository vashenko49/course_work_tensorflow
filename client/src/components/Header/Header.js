import React, { Component } from 'react';
import AppBar from '@material-ui/core/AppBar';
import Toolbar from '@material-ui/core/Toolbar';
import Typography from '@material-ui/core/Typography';
import Button from '@material-ui/core/Button';
import { bindActionCreators } from 'redux';
import * as UsersAction from '../../actions/User/Users';
import { connect } from 'react-redux';
import Auth from '../Auth/Auth';
import Box from '@material-ui/core/Box';
import './Header.scss';
import StyledLink from '../styled/StyledLink';

class Header extends Component {
  constructor(props) {
    super(props);
    this.state = {
      isOpenDrawer: false,
      isOpenDrawerAdmin: false
    };
  }

  render() {
    const { openWindowAuth, signOut } = this.props;
    const { isAuthorization } = this.props.Users;
    return (
      <AppBar position="static">
        <Toolbar>
          <StyledLink className={'admin-btn'} to={'/'} color="inherit">
            <Typography variant="h6">Home</Typography>
          </StyledLink>
          <Box className={'header-main-block'} display="flex" justifyContent="flex-end" m={1}>
            {isAuthorization ? (
              <Box>
                <Button onClick={signOut} color="inherit">
                  Sign Out
                </Button>
              </Box>
            ) : (
              <Box>
                <Button onClick={openWindowAuth} color="inherit">
                  Login
                </Button>
              </Box>
            )}
          </Box>
        </Toolbar>
        <Auth />
      </AppBar>
    );
  }
}

function mapStateToProps(state) {
  return { Users: state.Users };
}

function mapDispatchToProps(dispatch) {
  return {
    openWindowAuth: bindActionCreators(UsersAction.openWindowAuth, dispatch),
    signOut: bindActionCreators(UsersAction.signOut, dispatch)
  };
}

export default connect(mapStateToProps, mapDispatchToProps)(Header);
