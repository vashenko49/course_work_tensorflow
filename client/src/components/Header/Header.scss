.header-main-block{
  width: 100%;
}
.admin-btn{
  color: inherit;
}