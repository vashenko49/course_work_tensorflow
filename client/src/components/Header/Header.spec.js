import React from 'react';
import { createShallow } from '@material-ui/core/test-utils';
import { Header } from './Header';

describe('<header />', () => {
  let shallow;
  const props = {
    Users: {
      loading: true,
      openWindowLogIn: false,
      isAuthorization: false,
      jwt: '',
      error: '',
      personalInfo: {
        _id: '',
        isAdmin: false,
        login: '',
        firstName: '',
        lastName: '',
        middleName: ''
      }
    },
    openWindowAuth: () => {},
    signOut: () => {}
  };
  beforeAll(() => {
    shallow = createShallow();
  });

  it('should work', () => {
    const wrapper = shallow(<Header {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
