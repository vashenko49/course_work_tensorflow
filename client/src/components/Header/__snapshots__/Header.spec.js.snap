// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<header /> should work 1`] = `
<WithStyles(ForwardRef(AppBar))
  position="static"
>
  <WithStyles(ForwardRef(Toolbar))>
    <_default
      className="admin-btn"
      color="inherit"
      to="/"
    >
      <WithStyles(ForwardRef(Typography))
        variant="h6"
      >
        Home
      </WithStyles(ForwardRef(Typography))>
    </_default>
    <Styled(MuiBox)
      className="header-main-block"
      display="flex"
      justifyContent="flex-end"
      m={1}
    >
      <Styled(MuiBox)>
        <WithStyles(ForwardRef(Button))
          color="inherit"
          onClick={[Function]}
        >
          Login
        </WithStyles(ForwardRef(Button))>
      </Styled(MuiBox)>
    </Styled(MuiBox)>
  </WithStyles(ForwardRef(Toolbar))>
  <Connect(Auth) />
</WithStyles(ForwardRef(AppBar))>
`;
